lib/core-classes/DeviceService.ts

```diff
@@ -84,101 +60,158 @@ export class DeviceService {
-   * automatically attached to the one provided.
-   *
-   * If an assetId is provided, then the device will be linked to this asset.
-   *
-   * @param engineId If it's not the admin engine, then the device will be attached to this one
    * @param deviceContent Content of the device to create
+   * @param {Object} options
+   * @param options.engineId If it's not the admin engine,
+   *  then the device will be attached to this one
+   * @param options.measures Measures to be pushed to the device.
+   *  They will be filtered by name and date like an update
+   * @param options.linkRequest If specified, link to an asset
+   * @param options.refresh Wait for ES indexation
+   * @param options.strict If true, throw if an operation isn't possible
    */
   async create (
-    engineId: string,
     deviceContent: DeviceContent,
-    assetId: string,
-    { refresh }: { refresh?: any },
-  ): Promise<KDocument<DeviceContent>> {
-    const deviceId = Device.id(deviceContent.model, deviceContent.reference);
-
-    const device = await this.sdk.document.create<DeviceContent>(
-      this.config.adminIndex,
-      'devices',
-      deviceContent,
-      deviceId,
-      { refresh });
+    { engineId, measures, linkRequest, refresh, strict }: {
+      engineId?: string,
+      measures?: MeasureContent[],
+      linkRequest?: LinkRequest,
+      refresh?: any,
+      strict?: boolean
+    } = {},
+  ): Promise<Device> {
+    const device = new Device(
+      deviceContent, Device.id(deviceContent.model, deviceContent.reference));
 
-    if (this.config.adminIndex === engineId) {
-      return device;
+    if (measures) {
+      device.updateMeasures(measures);
     }
 
-    const attachedDevice = await this.attachEngine(
-      { deviceId, engineId },
-      { refresh, strict: true });
+    let asset: BaseAsset;
+    if (linkRequest) {
+      try {
+        asset = await this.assetService.getAsset(engineId, linkRequest.assetId);
+        device.linkToAsset(linkRequest);
+        asset.linkToDevice(linkRequest);
+      }
+      catch (error) {
+        if (strict) {
+          throw error;
+        }
+      }
+    }
 
-    if (! assetId) {
-      return attachedDevice;
+    if (engineId && this.config.adminIndex !== engineId) {
+      if (! await this.tenantExists(engineId)) {
+        if (strict) {
+          throw new BadRequestError(`Engine "${engineId}" does not exists.`);
+        }
+      }
+      else {
+        device._source.engineId = engineId;
+
+        await this.sdk.document.create(
+          engineId,
+          InternalCollection.DEVICES,
+          device._source,
+          device._id,
+          { refresh });
+      }
     }
 
-    const { device: linkedDevice } = await this.linkAsset({ assetId, deviceId }, { refresh });
+    await this.sdk.document.create(
+      this.config.adminIndex,
+      InternalCollection.DEVICES,
+      device._source,
+      device._id,
+      { refresh });
```

Comment:
double create!

Reply:
In facts it's not. It creates in the first one in the engine. The other create in the `device-manager`:`devices` that contains all devices